web-common/src/features/entity-management/WatchResourcesClient.ts

```diff
@@ -187,7 +187,8 @@ export class WatchResourcesClient {
             // If it's a new source, show the "Source imported successfully" modal
             const isNewSource =
               res.name.kind === ResourceKind.Source &&
-              res.resource.meta.specVersion === "1" &&
+              res.resource.meta.specVersion === "1" && // First file version
+              res.resource.meta.stateVersion === "2" && // First ingest is complete
```

Comment:
Is this `2` because `1` will be when it is queued for reconcile?

Also would it always be spec=1 and state=2 when the source gets successfully ingested for the 1st time? Like if the source had an error, the user fixed it in later iterations?

Reply:
> Is this 2 because 1 will be when it is queued for reconcile?

Exactly

> Also would it always be spec=1 and state=2 when the source gets successfully ingested for the 1st time? Like if the source had an error, the user fixed it in later iterations?

If the user later fixed it in-editor, then spec != 1. In this case, if we wanted to show a "source ingested for the first time" pop-up, I'd argue it should not be a full-screen blocking pop-up, but rather a less obstructive toast notification. We also have the top-right "Create Dashboard" CTA that is useful here.